modyn/models/gpt2/gpt2.py

```diff
@@ -0,0 +1,116 @@
+from typing import Any
+
+import torch
+from torch import nn
+from transformers import AutoTokenizer, GPT2Config, GPT2LMHeadModel
+
+from modyn.models.coreset_methods_support import CoresetSupportingModule
+
+
+class Gpt2:
+    # pylint: disable-next=unused-argument
+    def __init__(self, hparams: Any, device: str, amp: bool) -> None:
+        self.model = Gpt2Modyn(hparams)
+        self.model.to(device)
+
+
+"""
+Adapted from an example implementation of a GPT-2 model.
+This implementation uses the GPT-2 tokenizer from Hugging Face's Transformers library:
+https://huggingface.co/docs/transformers/model_doc/gpt2
+"""
+
+
+class Gpt2Modyn(CoresetSupportingModule):
+    def __init__(self, hparams: Any) -> None:
+        super().__init__()
+
+        # Use hparams to decide the GPT-2 version
+        model_name = hparams.model_name_or_path if hasattr(hparams, "model_name_or_path") else "gpt2-large"
```

Comment:
I don't get that. Isn't `model_name_or_path` super generic? Then the Gpt2 class would set this always to gpt2-large, but this reads more like a general purpose huggingface transformer class